tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -776,14 +776,16 @@ def with_integration_base_java_and_normalization(context: PipelineContext, build
     )
 
 
-async def with_airbyte_java_connector(context: ConnectorContext, connector_java_tar_file: File, build_platform: Platform) -> Container:
+async def with_airbyte_java_connector(
+    context: ConnectorContext, connector_java_tar_file: File, build_platform: Platform, bust_cache: bool = False
+) -> Container:
     application = context.connector.technical_name
-
     build_stage = (
         with_integration_base_java(context, build_platform)
         .with_workdir("/airbyte")
         .with_env_variable("APPLICATION", context.connector.technical_name)
-        .with_file(f"{application}.tar", connector_java_tar_file)
+        .with_env_variable("CACHEBUSTER", str(uuid.uuid4()) if bust_cache else "")
```

Comment:
non-blocking question - if every java build makes a new cache 'namespace' with this random UUID... do we ever clean up this chache?  It seems like it could get very large

Reply:
My understanding is that once the env var is set all the layers afterward are busted. I believe that once busted the layers are not cached anymore and the garbage collection logic is on the build kit side.

Reply:
And, even if the layers are not garbage collected, this change won't introduce uuid namespaced cache as layers are not namespaced and are reused / shared across images, so we're not introducing an ever growing cache.